server/lib/npm_installer.coffee

```diff
@@ -0,0 +1,89 @@
+path = require 'path'
+fs = require 'fs'
+directory = require './directory'
+executeUntilEmpty = require '../helpers/executeUntilEmpty'
+config = require('./conf').get
+log = require('printit')
+    date: true
+    prefix: 'lib:npm-installer'
+
+BASE_PACKAGE_JSON = """
+    {
+      "name": "cozy-controller-fake-package.json",
+      "version": "1.0.0",
+      "description": "This file is here to please NPM",
+      "README":  "This file is here to please NPM",
+      "license": "N/A",
+      "repository": "N/A"
+    }
+"""
+
+createAppFolder = (app, callback) ->
+    dirPath = path.join config('dir_app_bin'), app.name
+    packagePath = path.join dirPath, 'package.json'
+    fs.mkdir dirPath, "0700", (err) ->
+        if err then return callback new Error """
+            Failed to create folder #{dirPath} : #{err.message}
+        """
+        fs.writeFile packagePath, BASE_PACKAGE_JSON, 'utf8', (err) ->
+            if err then return callback new Error """
+                Failed to create package.json #{packagePath} : #{err.message}
+            """
+            directory.changeOwner app.user, dirPath, (err) ->
+                if err then return callback new Error """
+                    Failed to changeOwner #{dirPath} : #{err.message}
+                """
+                callback null, dirPath
+
+###
+    Initialize repository of <app>
+        * Run npm install <app> in the apps dir
+###
+module.exports.init = (app, callback) ->
+    unless app.package?.name
+        return callback new Error """
+            Tried to npm_installer.init a non NPM app : #{JSON.stringify app}
+        """
+
+    console.time "npm installing"
+    createAppFolder app, (err, appFolder) ->
+        return callback err if err
```

Comment:
In case of error, you should probably want to stop the `console.time`:

```coffee
if err
    console.timeEnd "npm install failed"
    return callback err
```